src/components/codeBlock.ts

```diff
@@ -23,10 +23,11 @@ export const inlineCodeBlock = (code: string, language: string) =>
   codeBlock(` ${code} `, language);
 
 export const multiLineCodeBlock = (code: string, language: string) => {
-  const maxLineChars = Math.max(...code.split("\n").map((line) => line.length));
+  const codeSplit = code.split("\n");
+
+  const maxLineChars = codeSplit.reduce((acc,curr)=> curr.length > acc ? curr.length : acc, 0);
```

Comment:
I prefer not to use reduce if possible. 
In that case, I think it's less obvious to see that it's finding the maximum than `Math.max`

Reply:
@yoavbls I agree it's less obvious, but it saves you 2 loops and won't take as much memory as spreading an array in Math.max. I think any VScode plugin should focus on these small gains.

Reply:
It seems like a pre-mature optimization to me but since we didn't measure it and there is someone in the issues that feels performance impact by the extension I agree.
Can you just add a comment saying this line is finding the longest line please?

Reply:
@yoavbls done

Reply:
just after a brief look at the code I would guess that all the regexps have biggest performance impact. Caching them should likely improve things a lot.

Reply:
I think it's the whole point of this package to do "replace all" :)

But I think the performance would improve a lot if those regexps' wouldn't be recreated on every invocation of the function and instead would be created in global scope (if those are not dynamic).

It should also help to combine regexes' to have less in total but that might not be feasible and it would probably make the code less readable.

Reply:
You're probably right. We have to find a way to do fewer replacements like this and invoke prettier just when it's necessary. 
@rchl @scr2em any help in measuring and solving this problem will be appreciated 🙏🏼